integrations/gcp/src/commonMain/kotlin/com/xebia/functional/xef/gcp/GcpChat.kt

```diff
@@ -0,0 +1,109 @@
+package com.xebia.functional.xef.gcp
+
+import com.xebia.functional.tokenizer.EncodingType
+import com.xebia.functional.tokenizer.ModelType
+import com.xebia.functional.xef.llm.Chat
+import com.xebia.functional.xef.llm.Completion
+import com.xebia.functional.xef.llm.models.chat.*
+import com.xebia.functional.xef.llm.models.text.CompletionChoice
+import com.xebia.functional.xef.llm.models.text.CompletionRequest
+import com.xebia.functional.xef.llm.models.text.CompletionResult
+import com.xebia.functional.xef.llm.models.usage.Usage
+import io.ktor.util.date.*
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.uuid.UUID
+import kotlinx.uuid.generateUUID
+
+@OptIn(ExperimentalStdlibApi::class)
+class GcpChat(apiEndpoint: String, projectId: String, modelId: String, token: String) :
+  Chat, Completion, AutoCloseable {
+  private val client: GcpClient = GcpClient(apiEndpoint, projectId, modelId, token)
+
+  override val name: String = client.modelId
+  override val modelType: ModelType =
+    ModelType.LocalModel(client.modelId, EncodingType.CL100K_BASE, 4096)
```

Comment:
The model used in the example only has 2048 as max tokens. I don't know if is better to set here the value. I don't know if is possible to modify from another place: 
```suggestion
    ModelType.LocalModel(client.modelId, EncodingType.CL100K_BASE, 2048)
```